resources/test/mock-ad-units.js

```diff
@@ -1,63 +1,269 @@
-/**
- * This file defines all ad units used on the site. The format of the config is:
- *
- *  '<ad unit name>': {
- *    'slotName': <name>,
- *    'sizes': <sizes>,
- *    'onSlotRenderEnded': <function (event, element)>
- *  }
- *
- * Where:
- *  - {string} name - Name that slot is referenced by when loading ads.
- *  - {array} sizes - attribute is a list of ad sizes, which can be passed
- *      directly to `slot.defineSizeMapping()` (https://developers.google.com/doubleclick-gpt/reference#googletag.Slot_defineSizeMapping).
- *  - {function} onSlotRenderEnded - function that takes an event and element,
- *      where element is the ad container. Fires when an ad slot has finished rendering.
- */
-
-module.exports = {
+var CLASS_SMALL_AD = 'small-ad';
+var CLASS_AD_PLUS_CONTENT = 'ad-plus-content';
 
+var AdUnits = {
   settings: {
     dfpSite: 'onion',
-    filterSlotsByViewport: 100
-  },
-
-  units: {
-
-    'header': {
-      slotName: 'header',
-      sizes: [
-        [[970, 0], [[728, 90], [970, 250], [970, 90], [970, 415]]],
-        [[728, 0], [728, 90]],
-        [[0, 0], [320, 50]]
-      ]
-    },
-
-    'article-header': {
-      slotName: 'header',
-      sizes: [
-        [[970, 0], [[728, 90], [970, 250]]],
-        [[728, 0], [728, 90]],
-        [[0, 0], [320, 50]]
-      ]
-    },
-
-    'sidebar-primary': {
-      sizes: [
-        [[0, 0], [300, 250]]
-      ]
-    },
-
-    'sidebar-secondary': {
-      sizes: [
-        [[0, 0], [300, 250]]
-      ]
-    },
-
-    'sidebar-tertiary': {
-      sizes: [
-        [[0, 0], [300, 250]]
-      ]
+    closeDelay: 1250
+  },
+
+  makeAdCloseable: function(adElement) {
+    var closeButton = document.createElement('div');
+    if (this.settings.closeDelay) {
+      closeButton.className = 'close-btn disabled';
+      setTimeout(function() {
+        utils.removeClass(closeButton, 'disabled');
+      }, this.settings.closeDelay);
+    } else {
+      closeButton.className = 'close-btn';
+    }
+    utils.addClass(adElement.parentElement, 'pinned');
+    adElement.appendChild(closeButton);
+
+    if('ontouchend' in document.documentElement) {
+      closeButton.addEventListener('touchend', AdUnits.closeAd.bind(null, adElement), false);
+    } else {
+      closeButton.addEventListener('click', AdUnits.closeAd.bind(null, adElement), false);
+    }
+  },
+
+  closeAd: function(adElement) {
+    if (onionan) {
```

Comment:
Same here, not immediately clear where `onionan` comes from. Also, we should start naming this something more verbose like `onionAnalytics`.